### DISCUSSION TITLE
How to use the `exit` api?

### DISCUSSION THREAD

**Discussion by junyoung-lee-1201** [tauri-apps/tauri#8612] [Q&A]
I want to execute some function when I close the app. Here's my approach:

```typescript
import { exit } from '@tauri-apps/api/process';
window.getCurrent().listen(TauriEvent.WINDOW_CLOSE_REQUESTED, async () => {
  invoke('test_command');
  await exit(0);
});
```

`test_command` is successfully invoked, but `exit(0)` doesn't seem to work. What am I doing wrong?

**Comment by FabianLars** [comment]
try appending `.catch(console.error)` to the exit(0) call and check the devtools console for errors.

**Comment by junyoung-lee-1201** [comment]
The error was happening because I didn't enable the process API in Tauri configuration. Adding it to `allowlist` solved it. Thanks!
